### ISSUE TITLE
TypeError thrown by document_table in getWhereForDoc

### ISSUE THREAD

**Issue by jadams74** [robconery/massive-js#300]
I have run into TypeError being throw when evaluating the conditions == "*" line in getWhereForDoc. Conditions here is set to the defaults found in arg_types.js. When I examine the conditions object, it is just __prototype__, nothing else there. This seems consistent with the behavior lined out in arg_types.findArgs(). 

Should this (document_table.js line 88:34 )be a non-coercive call? 

Where can we set these conditions?  I don't see it as an argument to document_table.js.findDoc().

**Comment by robconery** [comment]
I hate to say I have no idea what you're talking about :/ can you show me your failing code and the error you're getting? I'll do my best to help.

**Comment by jadams74** [comment]
We've updated several of our project's dependencies. Upon doing so line 99 in document_table.js started throwing a TypeError. Here is the stack trace:

```
160809/135207.679, [error] message: Uncaught error: Cannot convert object to primitive value stack: TypeError: Uncaught error: Cannot convert object to primitive value
    at __dirname.getWhereForDoc (/home/jadams/projects/generators/generator-samples/hapi-app/todo-list/app/node_modules/massive/lib/document_table.js:88:34)
    at exports.findDoc (/home/jadams/projects/generators/generator-samples/hapi-app/todo-list/app/node_modules/massive/lib/document_table.js:76:20)
    at Object.services.find (tasks.js:84:24)
    at handler (task-find-get-config.js:25:63)
    at Object.internals.handler (/home/jadams/projects/generators/generator-samples/hapi-app/todo-list/app/node_modules/hapi/lib/handler.js:96:36)
    at Items.serial (/home/jadams/projects/generators/generator-samples/hapi-app/todo-list/app/node_modules/hapi/lib/handler.js:63:26)
    at done (/home/jadams/projects/generators/generator-samples/hapi-app/todo-list/app/node_modules/hapi/node_modules/items/lib/index.js:31:25)
    at done (/home/jadams/projects/generators/generator-samples/hapi-app/todo-list/app/node_modules/hapi/node_modules/items/lib/index.js:63:25)
    at pre (/home/jadams/projects/generators/generator-samples/hapi-app/todo-list/app/node_modules/hapi/lib/handler.js:49:28)
    at finalize (/home/jadams/projects/generators/generator-samples/hapi-app/todo-list/app/node_modules/hapi/lib/handler.js:281:20)
    at wrapped (/home/jadams/projects/generators/generator-samples/hapi-app/todo-list/app/node_modules/hapi/node_modules/hoek/lib/index.js:872:20)
    at internals.Response._processPrepare (/home/jadams/projects/generators/generator-samples/hapi-app/todo-list/app/node_modules/hapi/lib/response.js:457:16)
    at internals.Response._prepare (/home/jadams/projects/generators/generator-samples/hapi-app/todo-list/app/node_modules/hapi/lib/response.js:396:21)
    at process.nextTick (/home/jadams/projects/generators/generator-samples/hapi-app/todo-list/app/node_modules/hapi/lib/reply.js:151:22)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
```

The code in question is the second part of the OR statement:

`|| conditions == "*")`

I think this should be:

`|| conditions === "*")`

The change does solve my problem. However, I am not a javascript expert and wanted to call it out here. There may be a legitimate reason why you chose the coercive syntax. Note that by changing it to ===, the code evaluates to false as I would expect. It does not attempt to convert {} to a string.

'conditions' itself is populated in arg_types.js.searchArgs, where the default is set to {}. That is the value the 'conditions variable' that is causing the error.  I would also like to know how to set that value instead of taking the Default. This might also help me solve the problem without requiring a change here.

**Comment by robconery** [comment]
I do appreciate you troubleshooting this for me, however I'm still very unclear on *what the problem is*. I can't tell from the code you've provided what is actually *triggering* this error.

Let's try this again.

You're passing Massive *something* and I need to know what that is. Specifically: **what is the API call you're making**. I can tell that `tasks` is your document table and you're trying to find something, but without knowing the criteria (and options) it's impossible to troubleshoot this.

**Comment by jadams74** [comment]
we are trying to find all tasks with the query {} and setting limit and offset.

```
var query = {};
var options = {
  limit:10,
  offset: 0
};

db.tasks.findDoc(query, options, function (err, tasks)  {
    // do something
});
```

I did some more work on this today. Turns out upgrading hapi.js from 11.x to 12.+ causes the condition to manifest.  For now we are staying at hapi 11.x till I get this figured out.  Also, I am currently putting together a hapi project using massive to attempt to reproduce minus the complexity of the rest of our project. I am having no luck here either. I have no idea what code state is actually triggering the error :/
